src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/ThreadPoolTaskScheduler.cs

```diff
@@ -45,13 +45,11 @@ protected internal override void QueueTask(Task task)
             if (Thread.IsThreadStartSupported && (options & TaskCreationOptions.LongRunning) != 0)
             {
                 // Run LongRunning tasks on their own dedicated thread.
-#pragma warning disable CA1416 // TODO: https://github.com/dotnet/runtime/issues/44922
```

Comment:
Cannot get rid of the warnings in mono build. My best guess is the linker is trimming the attribute, so now I do not know what to do with it CC @eerhardt @marek-safar

Reply:
> My best guess is the linker is trimming the attribute

No, you need to also annotate 

https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/mono/System.Private.CoreLib/src/System/Threading/Thread.Browser.Mono.cs#L10

You should actually add `SupportedOSPlatformGuardAttribute` https://github.com/dotnet/runtime/blob/main/src/mono/System.Private.CoreLib/src/ILLink/ILLink.LinkAttributes.xml#L96 to be trimmed in the final apps

Reply:
> My guess is the analyzer could have problem with const field

@marek-safar that was absolutely right, somehow flow analysis treating const fields differently, probably because it is compile-time constant, if its value is true it affects entire call-site without evaluating `true/false` side of `if block`. if its value is false entire block is evaluated false even its value checked within a `if block`. I guess it is not a bug, might be by design, could we use readonly property instead?

Reply:
> could we use readonly property instead?

Yep, that's fine.